src/sentry/migrations/0678_add_query_sub_id_to_aar_activations.py

```diff
@@ -0,0 +1,52 @@
+# Generated by Django 5.0.2 on 2024-03-21 18:46
+
+from django.db import migrations, models
+
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
+
+    dependencies = [
+        ("sentry", "0677_unpickle_project_options_again"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            database_operations=[
+                migrations.RunSQL(
+                    """
+                    ALTER TABLE "sentry_alertruleactivations" ADD COLUMN "query_subscription_id" integer NOT NULL DEFAULT 0;
```

Comment:
Shouldn't this be a nullable column? It sounds like an optional column/relation.

Reply:
Ah
so - actually i _do_ want this to be required since activations will always be tied to a query_subscription as well (but the query subscription's can be deleted and we want this instance to persist still)